activemodel/lib/active_model/secure_password.rb

```diff
@@ -1,18 +1,18 @@
 # frozen_string_literal: true
 
+require "active_model/secure_password/base"
+
 module ActiveModel
   module SecurePassword
     extend ActiveSupport::Concern
 
-    # BCrypt hash function can handle maximum 72 bytes, and if we pass
-    # password of length more than 72 bytes it ignores extra characters.
-    # Hence need to put a restriction on password length.
-    MAX_PASSWORD_LENGTH_ALLOWED = 72
-
     class << self
       attr_accessor :min_cost # :nodoc:
+      attr_accessor :available_password_algorithms # :nodoc:
     end
     self.min_cost = false
+    # TODO: is this the best pattern to discover/register password hashing algo classes?
```

Comment:
I'd say the "Rails way" would be to make this configurable so one can write something like

```
config.active_model.secure_password_algorithms << :some_algo
```

or 

```
config.active_model.secure_password_algorithms << 'SomeAlgo'
```

The default value of that option would then be just all Algorithms that are supported by default (only BCrypt for now?).

An example of how to add configurable options can be seen in: https://github.com/rails/rails/pull/41158 (or just search for something like "add config" in the commit history, there are plenty of examples)